.github/workflows/flake-update-lockfile.yml

```diff
@@ -0,0 +1,36 @@
+name: Update flake lockfile
+# adapted from
+# https://github.com/numtide/system-manager/blob/82d5a9ecd15ec48bcbfbacf5462066ee267d6aae/.github/workflows/update-flake-lock.yml
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "45 8 * * 1" # runs weekly every Monday at 08:45 UTC/03:45 ET
+  push:
+    branches: []
+
+jobs:
+  lockfile:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "Checkout the repo"
+        uses: "actions/checkout@v4"
+      - name: "Install the Nix package manager"
+        uses: "cachix/install-nix-action@master"
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Update the lockfile"
+        run: |
+          # setup git user
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+          # run equivalent of `nix flake update`
+          nix --option commit-lockfile-summary "chore: update flake.lock" \
+          flake update --no-use-registries --commit-lock-file
```

Comment:
Following up on https://github.com/ALT-F4-LLC/kickstart.nix/issues/96#issuecomment-2677235503, I should note that there are all three of 

- user
- system
- global

flake registries. What we've done with #100 is to disable the system-level but the user and global registries may still be accessible depending on the nix configuration. The workflow here is more explicit about also disabling the user and global registries as well with the `--no-use-registries` flag. The upstream discussion of this remains in https://github.com/NixOS/nix/issues/7422 and elsewhere.